src/khoj/processor/conversation/gpt4all/utils.py

```diff
@@ -35,7 +37,14 @@ def download_model(model_name):
                 for chunk in r.iter_content(chunk_size=8192):
                     f.write(chunk)
                     progress_bar.update(len(chunk))
+
+        logger.debug(f"Successfully downloaded model {model_name} from {url} to {tmp_filename}")
+        # Move the tmp file to the actual file
```

Comment:
We should add a comment somewhere here to explain why the partial file is downloaded to a temp file path first (i,e to prevent partial corrupted downloads being used)